pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -80,7 +87,24 @@ func (u UsernamePassword) Setup(baseDir string) (*url.URL, error) {
 	return overrideSourceURL, nil
 }
 
-func doSetup(sourceURL url.URL, usernameSecret, passwordSecret, tokenSecret string) (*url.URL, *url.URL, error) {
+type overrideURLFunc func(sourceURL *url.URL, username, password string) (*url.URL, error)
+
+func removeCredentials(sourceURL *url.URL, username, password string) (*url.URL, error) {
+	overrideURL := *sourceURL
+	overrideURL.User = nil
+	return &overrideURL, nil
+}
+
+func longPasswordOverride(dir string, overrideFn overrideURLFunc) overrideURLFunc {
+	return func(sourceURL *url.URL, username, password string) (*url.URL, error) {
+		if len(password) > curlPasswordThreshold {
+			return startProxy(dir, sourceURL, username, password)
```

Comment:
is there a significant downside to just always using the proxy path, rather than having two paths to test?  especially since we're always going to be in the proxy path today due to the token length?